### ISSUE TITLE
Enable 16bit precision on RNN models

### ISSUE THREAD

**Issue by ant0nsc** [microsoft/InnerEye-DeepLearning#378]
With the move to PL, some of the models at present only support 32bit precision. When running them in 16bit precision, it says `torch.nn.functional.binary_cross_entropy and torch.nn.BCELoss are unsafe to autocast. Many models use a sigmoid layer right before the binary cross entropy layer. In this case, combine the two layers using torch.nn.functional.binary_cross_entropy_with_logits or torch.nn.BCEWithLogitsLoss.  binary_cross_entropy_with_logits and BCEWithLogits are safe to autocast.`

**Comment by ant0nsc** [comment]
Fixed with PR #402
